feat(spanner): support for enum columns as keys, index and integration tests

harshachinta opened the pull request:
This PR has the following changes,
1. Adds support for Enum columns in primary key and indexes.
2. Integration tests for Parameter Binding.
3. Integration tests for Proto fields and Enum columns as Primary keys.
4. Integration tests for Proto fields and Enum columns as Indexes.

olavloite commented on spanner/integration_test.go:
```diff
@@ -2317,6 +2317,165 @@ func TestIntegration_BasicTypes_ProtoColumns_Errors(t *testing.T) {
+			want: [][]interface{}{{int64(3), "Singer3", "Singer3"}},
+		},
+	} {
+		ro := client.ReadOnlyTransaction()
```
This (probably) causes a session leak, as this read-only transaction is never closed.

Shua1 replied:
Are these comments addressed?

harshachinta replied:
Yes @Shua1. The changes have already been committed.

olavloite commented on spanner/integration_test.go:
```diff
@@ -2317,6 +2317,165 @@ func TestIntegration_BasicTypes_ProtoColumns_Errors(t *testing.T) {
+	}
+
+	// Read all rows based on Proto Enum Primary key column values
+	ro := client.ReadOnlyTransaction()
```
Same here, not closing a read-only transaction will cause a session leak.

Shua1 reviewed (commented):
1. Please split the parameter binding part.
2. It seems more accurate, in the descriptor of the PR, to say "support for enum columns..." instead of "support for proto columns".

harshachinta commented:
> 1. Please split the parameter binding part.
> 2. It seems more accurate, in the descriptor of the PR, to say "support for enum columns..." instead of "support for proto columns".

1. Updated the PR description to make the changes more clearer. As the test for parameter binding is small, raising a separate PR for that would not be useful.
2. Updated the title of the PR.

Shua1 reviewed (commented):
The PR description says: "support for enum columns as keys, index, parameters".

But we are not adding support for parameters. Right? Please update.

harshachinta commented:
> The PR description says: "support for enum columns as keys, index, parameters".
> 
> But we are not adding support for parameters. Right? Please update.

Updated the description. Thanks for the review comment, @Shua1
